### PR TITLE
Post-workshop fixes

### PR DESCRIPTION
**Pull request by nsoranzo**
The following fixes result from running this tutorial during a [Software Carpentry Workshop at the Earlham Institute](http://earlham.ac.uk/software-carpentry-workshop-2017) together with @froggleston and @tonietuk.

See individual commits for details.

### CODE REVIEW

File: _episodes/08-defensive.md [hunk 1]
```diff
@@ -531,11 +531,13 @@ This violates another important rule of programming:
 > >
 > > def range_overlap(ranges):
 > >     '''Return common overlap among a set of [low, high] ranges.'''
+> >     if not ranges:
```
Comment (tbekolay) [inline _episodes/08-defensive.md]: What do you think about `if len(ranges) == 0` instead? I know experienced Python users like the simplicity of this, but to a novice it feels more clear that an empty list has length 0.

File: _episodes/02-loop.md [hunk 2]
```diff
@@ -355,8 +356,8 @@ so we should always use it when we can.
 > ## Computing the Value of a Polynomial
 >
 > The built-in function `enumerate` takes a sequence (e.g. a list) and generates a
-> new sequence of the same length. Each element of the new sequence contains the index
-> (0,1,2,...) and the value from the original sequence:
+> new sequence of the same length. Each element of the new sequence is a pair composed by the index
```
Comment (tbekolay) [inline _episodes/02-loop.md]: Nitpick: "composed by" -> "composed of"

File: _episodes/08-defensive.md (hunk 1 above)
Comment (nsoranzo) [inline _episodes/08-defensive.md]: On the other hand `not ranges` also handles the case where `ranges` is `None`.

Comment (tbekolay) [inline _episodes/08-defensive.md]: This is true, but not a situation I suspect our novice students will run into as I'm not sure we ever talk about `None` in the lesson. The more likely situation is they call it and pass nothing, which raises a `TypeError`. That said, if you strongly prefer to keep it this way I can probably live with it.

Comment (nsoranzo) [inline _episodes/08-defensive.md]: We talk about `None` a bit before this exercise:
```
`None` is built into Python, and means "nothing here".
```

What about I add a comment:
```python
    if not ranges:
        # ranges is None or an empty list
        return None
```

Comment (tbekolay) [inline _episodes/08-defensive.md]: > What about I add a comment

Yeah, that sounds great, thanks!

**Comment by tbekolay** [comment]
Thanks for the improvements!

**Pull request merged**